### PR TITLE
Add support dev cluster

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
Add a flag `dev` to flytectl demo start. It's used to start a dev cluster (minio + Postgres). Therefore, users are able to start a flyte cluster via the single binary locally and save data to minio or Postgres.

Entire workflow will be like
```bash
flytectl demo start --dev --image pingsutw/sandbox-lite-test
flyte start --config flyte_local.yaml # I've updated local config in flyte repo
```
cc @hamersaw Let me know if I missed anything
## Type
- [ ] Bug Fix
- [x] Feature
- [ ] Plugin

## Are all requirements met?

- [x] Code completed
- [ ] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
I've tested it by running a workflow.
image

## Tracking Issue
^^^

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by pingsutw** [comment]
blocked by https://github.com/flyteorg/flyte/pull/2891

**Review by hamersaw — APPROVED** [review]
